stl/inc/__msvc_chrono.hpp

```diff
@@ -666,18 +666,38 @@ namespace chrono {
         using time_point                = _CHRONO time_point<steady_clock>;
         static constexpr bool is_steady = true;
 
+#if defined(_M_ARM) || defined(_M_ARM64) // vvv ARM or ARM64 arch vvv
+#define _LIKELY_ARM _LIKELY
+#define _LIKELY_X86
+#elif defined(_M_IX86) || defined(_M_X64) // ^^^ ARM or ARM64 arch / x86 or x64 arch vvv
+#define _LIKELY_ARM
+#define _LIKELY_X86 _LIKELY
+#else // ^^^ x86 or x64 arch / other arch vvv
+#define _LIKELY_ARM
+#define _LIKELY_X86
+#endif // ^^^ other arch ^^^
         _NODISCARD static time_point now() noexcept { // get current time
             const long long _Freq = _Query_perf_frequency(); // doesn't change after system boot
             const long long _Ctr  = _Query_perf_counter();
             static_assert(period::num == 1, "This assumes period::num == 1.");
-            // 10 MHz is a very common QPC frequency on modern PCs. Optimizing for
-            // this specific frequency can double the performance of this function by
-            // avoiding the expensive frequency conversion path.
-            constexpr long long _TenMHz = 10'000'000;
-            if (_Freq == _TenMHz) {
+            // The compiler recognizes the constants for frequency and time period and uses shifts and
+            // multiplies instead of divides to calculate the nanosecond value.
+            constexpr long long _TwentyFourMHz = 24'000'000;
+            constexpr long long _TenMHz        = 10'000'000;
+            // clang-format off
+            if (_Freq == _TenMHz) _LIKELY_X86 {
+                // 10 MHz is a very common QPC frequency on modern x86 PCs. Optimizing for
+                // this specific frequency can double the performance of this function by
+                // avoiding the expensive frequency conversion path.
                 static_assert(period::den % _TenMHz == 0, "It should never fail.");
                 constexpr long long _Multiplier = period::den / _TenMHz;
                 return time_point(duration(_Ctr * _Multiplier));
+            } else if (_Freq == _TwentyFourMHz) _LIKELY_ARM {
+                // 24 MHz frequency is a common frequency on ARM64, including cases where it emulates x86
+                // (Windows devices, and Apple Silicon Macs using Parallels Desktop)

```

Comment:
I think we should say "ARM/ARM64", the part about emulation is potentially confusing so I think we should just drop it, and I'd especially like to avoid mentioning specific vendors.

Reply:
I like how it mentions emulation, it explains why that branch exists on x86/x64 and isn't under an ifdef or something.

I agree with removing the paren, though.

Reply:
Yeah, that's a good point. I'll put it back but generalize it to x86/x64.